File: mysql-async/src/main/scala/com/github/mauricio/async/db/mysql/blob/encoder/ByteArrayEncoder.scala
package com.github.mauricio.async.db.mysql.blob.encoder

import io.netty.buffer.{ByteBuf, Unpooled}

object ByteArrayEncoder extends BlobEncoder {

  def isLong(value: Any): Boolean = value.asInstanceOf[Array[Byte]].length > BlobEncoder.LONG_THRESHOLD

  def encode(value: Any): ByteBuf = Unpooled.wrappedBuffer(value.asInstanceOf[Array[Byte]])

}

File: mysql-async/src/main/java/com/github/mauricio/async/db/mysql/binary/encoder/ByteArrayEncoder.kt


package com.github.mauricio.async.db.mysql.binary.encoder

import com.github.jasync.sql.db.util.writeLength
import com.github.mauricio.async.db.mysql.column.ColumnTypes
import io.netty.buffer.ByteBuf

object ByteArrayEncoder : BinaryEncoder {
  override fun encode(value: Any, buffer: ByteBuf) {
    val bytes = value as ByteArray

    buffer.writeLength(bytes.size.toLong())
    buffer.writeBytes(bytes)
  }

  override fun encodesTo(): Int = ColumnTypes.FIELD_TYPE_BLOB

}
